python/lsst/cp/pipe/defects.py

```diff
@@ -238,10 +253,41 @@ def findHotAndColdPixels(self, exp, nSigma):
                 continue
 
             # Remove a background estimate
-            ampImg -= afwMath.makeStatistics(ampImg, afwMath.MEANCLIP, ).getValue()
+            meanClip = afwMath.makeStatistics(ampImg, afwMath.MEANCLIP, ).getValue()
+            ampImg -= meanClip
 
+            # Determine thresholds
+            stDev = afwMath.makeStatistics(ampImg, afwMath.STDEVCLIP, ).getValue()
+            expTime = exp.getInfo().getVisitInfo().getExposureTime()
+            thresholdType = self.config.thresholdType
+            datasetType = exp.getMetadata().get('IMGTYPE', 'UNKNOWN')
+            if thresholdType == 'VALUE':
+                hotPixelThreshold = self.config.darkCurrentThreshold*expTime/amp.getGain()
+                if datasetType.lower() == 'dark':
+                    nSigmaList = [hotPixelThreshold]
+                else:
+                    coldPixelThreshold = -(1.-self.config.fracThreshold)*meanClip
+                    nSigmaList = [hotPixelThreshold, coldPixelThreshold]
+                # Find equivalent sigma values.
+                nSigmaList = [x/stDev for x in nSigmaList]
+            else:
+                hotPixelThreshold = self.config.nSigmaBright
+                coldPixelThreshold = self.config.nSigmaDark
+                if datasetType.lower() == 'dark':
+                    nSigmaList = [hotPixelThreshold]
+                else:
+                    nSigmaList = [hotPixelThreshold, coldPixelThreshold]
+
+            # Go back to pixel values to provide log info below
+            valuesList = [np.array(nSigmaList[0])*stDev*amp.getGain()/expTime]
+            if len(nSigmaList) == 2:
+                valuesList.append(np.array(nSigmaList[1])*stDev/meanClip + 1.)
```

Comment:
I don't understand the above computations or why scalars are wrapped in an `np.array`.  We have the hot and cold pixel thresholds above, and maybe this is reverse engineering that, with very specific assumptions.  Or the calculations could be done in the `else` block above.  But they really are just "hot" and "cold" values and they are not the same so shouldn't be listed together.

Reply:
In the `if  thresholdType == 'VALUE'` ... `else` conditional, `nSigmaList` will always have the list of thresholds in `STDEV` units (with a bit of overloading of the name in the first part of the conditional). After the conditional block, we'd like to report in the log both the `STDEV` and equivalent pixel `VALUE` values, so we do that conversion.  We put things in arrays because, in principle, the thresholds are a list (hot a cold for flats, currently) and there is an iteration over that list/array in the algorithm below (we wanted to modify the current algorithm as little as possible).  As for the actual formulas from the thresholds, these come directly from the `eoTest` code. I can add a few more comments explaining the `eoTest` requirements, I think Jim has shared these with us somewhere.